src/highlighters.cc

```diff
@@ -1104,13 +1106,22 @@ struct LineNumbersHighlighter : Highlighter
         if (separator.length() > 10)
             throw runtime_error("separator length is limited to 10 bytes");
 
+        Optional<StringView> separator_cursor = parser.get_switch("separator-cursor");
+        if (separator_cursor && (*separator_cursor).length() != separator.length()) {
+            // Throw runtime error instead?
```

Comment:
Yep, I think that would make sense, also you could use `separator_cursor->column_length()` as its really the display length that we want to compare, not the byte one.